src/main/java/com/cinema/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.cinema.dao.impl;
+
+import com.cinema.dao.MovieDao;
+import com.cinema.lib.Dao;
+import com.cinema.model.Movie;
+import com.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't add movie " + movie.getTitle());
```

Comment:
you need to create and throw here your own exception

Reply:
I agree, let's create a custom DataProcessingException

Reply:
Done